app/models/miq_provision_request.rb

```diff
@@ -29,6 +29,10 @@ def self.request_task_class_from(attribs)
     vm_or_template = VmOrTemplate.find_by(:id => source_id)
     raise MiqException::MiqProvisionError, "Unable to find source Template/Vm with id [#{source_id}]" if vm_or_template.nil?
 
+    if vm_or_template.ext_management_system.nil?
```

Comment:
I've extracted it into a method to retrieve the vm/template or raise an exception if failed to.